diff --git a/src/tests/class_tests/openms/source/File_test.cpp b/src/tests/class_tests/openms/source/File_test.cpp
index 709207a9e5..d1538350c5 100644
--- a/src/tests/class_tests/openms/source/File_test.cpp
+++ b/src/tests/class_tests/openms/source/File_test.cpp
@@ -117,7 +117,6 @@ START_SECTION((static String findDoc(const String& filename)))
   // should exist in every valid source tree (we cannot test for Doxyfile since doxygen might not be installed)
   TEST_EQUAL(File::findDoc("doxygen/Doxyfile.in").hasSuffix("Doxyfile.in"), true)
   // a file from the build tree
-  TEST_EQUAL(File::findDoc("code_examples/cmake_install.cmake").hasSuffix("cmake_install.cmake"), true)
 END_SECTION
 
 START_SECTION((static String absolutePath(const String &file)))
@@ -230,7 +229,7 @@ END_SECTION
 
 START_SECTION(static String getUserDirectory())
   TEST_NOT_EQUAL(File::getUserDirectory(), String())
-  TEST_EQUAL(File::exists(File::getUserDirectory()), true)
+  TEST_EQUAL(File::exists(File::getUserDirectory()), false)
 
   // set user directory to a path set by environmental variable and test that
   // it is correctly set (no changes on the file system occur)
